src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -312,24 +327,43 @@ def _forward(self, model_inputs):
             out = {"tokens": tokens}
         else:
             stride = model_inputs.pop("stride", None)
-            input_values = model_inputs.pop("input_values")
+            input_values = (
+                model_inputs.pop("input_values")
+                if "input_values" in model_inputs
+                else model_inputs.pop("input_features")
+            )
             attention_mask = model_inputs.pop("attention_mask", None)
-            outputs = self.model(input_values=input_values, attention_mask=attention_mask)
+            outputs = self.model(input_values, attention_mask=attention_mask)
             logits = outputs.logits
 
             if self.type == "ctc_with_lm":
                 out = {"logits": logits}
             else:
                 out = {"tokens": logits.argmax(dim=-1)}
             if stride is not None:
+                # The stride is either a 3-tuple or a list of 3-tuples representing a batch
+                is_batched = not isinstance(stride, tuple)
+                batch_length = len(stride) if is_batched else 1
+
+                # If the proportion of input feature frames to logit frames can be statically
+                # determined based on convolution strides, use that. Otherwise, we need to
+                # calculate it here dynamically.
+                inputs_to_logits_ratio = getattr(self.model.config, "inputs_to_logits_ratio", None)
```

Comment:
Hmm this is a bit inconsistent with the line above where the default is `1` and not `None`

Reply:
@Narsil could you take a look here?

Reply:
Replied to your comment above to clarify the inconsistency but basically:

- Above we're assigning `align_to`, and if we don't know the inputs-to-logits-ratio, `1` is the only reasonable choice (that I can think of)
- But here we're actually rescaling the strides, and so if we don't know the inputs-to-logits-ratio, we'll need to calculate it